reporter/src/main/kotlin/reporters/OpossumReporter.kt

```diff
@@ -0,0 +1,599 @@
+                value.toFileList().map { resolvePath(key, it, isDirectory = false) }
+            }.plus("/")
+    }
+
+    data class OpossumFrequentLicense(
+        var shortName: String,
+        var fullName: String?,
+        var defaultText: String?
+    ) : Comparable<OpossumFrequentLicense> {
+        fun toJson(): Map<*, *> = sortedMapOf(
+            "shortName" to shortName,
+            "fullName" to fullName,
+            "defaultText" to defaultText
+        )
+
+        override fun compareTo(other: OpossumFrequentLicense) =
+            compareValuesBy(this, other,
+            { it.shortName },
+            { it.fullName },
+            { it.defaultText })
+    }
+
+    data class OpossumExternalAttributionSource(
+        var name: String,
+        var priority: Int,
+    )
+
+    data class OpossumInput(
+        var resources: OpossumResources = OpossumResources(),
+        var signals: MutableList<OpossumSignal> = mutableListOf(),
+        var pathToSignal: SortedMap<String, SortedSet<UUID>> = sortedMapOf(),
+        var packageToRoot: SortedMap<Identifier, SortedMap<String, Int>> = sortedMapOf(),
+        var attributionBreakpoints: SortedSet<String> = sortedSetOf(),
+        var filesWithChildren: SortedSet<String> = sortedSetOf(),
+        var frequentLicenses: SortedSet<OpossumFrequentLicense> = sortedSetOf(),
+        var baseUrlsForSources: SortedMap<String, String> = sortedMapOf(),
+        var externalAttributionSources: SortedMap<String, OpossumExternalAttributionSource> = sortedMapOf()
+    ) {
+        fun toJson(): Map<*, *> =
+            sortedMapOf(
+                "metadata" to sortedMapOf(
+                    "projectId" to "0",
+                    "fileCreationDate" to LocalDateTime.now().toString()
+                ),
+                "resources" to resources.toJson(),
+                "externalAttributions" to signals
+                    .map { it.toJson() }
+                    .flatMap { it.toList() }
+                    .toMap(),
+                "resourcesToAttributions" to pathToSignal.mapKeys {
+                    val trailingSlash = if (resources.isPathAFile(it.key)) "" else "/"
+                    resolvePath("${it.key}$trailingSlash")
+                },
+                "attributionBreakpoints" to attributionBreakpoints,
+                "filesWithChildren" to filesWithChildren,
+                "frequentLicenses" to frequentLicenses.toList().map { it.toJson() },
+                "baseUrlsForSources" to baseUrlsForSources,
+                "externalAttributionSources" to externalAttributionSources
+            )
+
+        fun getSignalsForFile(file: String): List<OpossumSignal> =
+            pathToSignal[file]
+                ?.mapNotNull { uuid -> signals.find { it.uuid == uuid } }
+                ?: emptyList()
+
+        fun addAttributionBreakpoint(breakpoint: String) {
+            attributionBreakpoints.add(resolvePath(breakpoint, isDirectory = true))
```

Comment:
Use the `+=` operator: `attributionBreakpoints += resolvePath(breakpoint, isDirectory = true)`